### PR TITLE
[AKS] `aks enable-addons`: Add new parameter `--enable-msi-auth-for-monitoring` to support enabling managed identity auth

### PR DESCRIPTION
**Pull request by ganga1980**
**Description**

This PR introduces additional flag --enable-msi-auth-for-monitoring for customer to enable AKS monitoring addon with managed identity auth 

**Testing Guide**

-  Enable monitoring addon with MSI auth for the new cluster
az aks create -g  `<rgName>` -n `<clusterName>` --enable-addons monitoring --enable-msi-auth-for-monitoring --generate-ssh-keys 
- Enable Monitoring addon with MSI auth for existing AKS cluster
az aks enable-addons -a monitoring -g `<rgName>` -n `<clusterName>` --enable-msi-auth-for-monitoring  

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@ganga1980 I modified the PR title, please see if it's description is accurate

**Comment by ganga1980** [comment]
Hi,  @jsntcy , @kairu-ms , I would like to merge this PR to meet our release schedule, can you please review this PR? I have tested all the scenarios and this PR up to date.

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # --------------------------------------------------------------------------------------------
 
 import importlib
+from tkinter import FALSE
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py:7]: This line breaks CI and it is never used:

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1464021&view=logs&j=964b1a32-a564-5964-77b9-8017280f6c54&t=fd408678-da66-5f65-344b-e09411c080cb

```
==================================== ERRORS ====================================
_______________ ERROR collecting tests/latest/test_decorator.py ________________
ImportError while importing test module '/usr/lib64/az/lib/python3.6/site-packages/azure/cli/command_modules/acs/tests/latest/test_decorator.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/usr/lib64/az/lib/python3.6/site-packages/azure/cli/command_modules/acs/tests/latest/test_decorator.py:7: in <module>
    from tkinter import FALSE
E   ModuleNotFoundError: No module named 'tkinter'
```

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py:7]: @ganga1980 May I ask why this unused import was added here?

Comment (ganga1980) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_decorator.py:7]: > @ganga1980 May I ask why this unused import was added here?

> @zhoxing-ms , Not sure why this got added automatically. please go-ahead and remove this unused import.

**Pull request merged**